packages/api2/src/praise/praise.service.ts

```diff
@@ -90,6 +91,63 @@ export class PraiseService {
     return praisePagination;
   }
 
+  /**
+   * returns all of the model in json format
+   * Do not populate relations
+   */
+  async export(
+    format: string = 'csv',
+    startDate: string,
+    endDate: string,
+    periodId: string,
+  ): Promise<Praise[] | string> {
+    const [fromDate, toDate] = await this.selectDateFilterRange(
+      periodId, startDate, endDate
+    );
+
+    const praises = await this.praiseModel
+      .find({
+        createdAt: { $gte: fromDate, $lte: toDate },
+      })
+      .lean();
+
+    if (format !== 'csv') return praises;
+    return parse(praises);
+  }
+
+  /**
+   * Select date range to filter by, using either period or custom dates
+   * @param {string} periodId
+   * @param {String} startDate
+   * @param {String} endDate
+   * @returns {Promise<[Date, Date]>}
+   * @throws {ServiceException}
+   *
+   **/
+  async selectDateFilterRange(
+    periodId: string,
+    startDate: string,
+    endDate: string,
+  ): Promise<[Date, Date]> {
+    if (periodId && startDate && endDate) {
```

Comment:
Query should fail if periodId and either startDate or endDate is specified.

Reply:
I added the response object to the controller, if the content type is csv I use content disposition.

```
res.set({
      'Content-Type': 'text/csv',
      'Content-Disposition': 'attachment; filename="users.csv"',
    });
 ``` 

I wasn't 100% sure what is the best way to do it in nestJs, I found that way

Reply:
Thanks!